### PR TITLE
refactor(zone.js): remove legacy browser support

### LINKED ISSUE #46773: Clarify IE11 support state for `zone.js` / and clean-up zone.js code/tests for IE11-specific logic
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

No

### Description

Angular no longer supports IE11, but it's not quite clear from the documentation whether `zone.js` is included in the angular.io browser guarantees, or if Zone has its own browser guarantees (given it also being versioned differently etc.)

If IE11 is no longer supported in ZoneJS either, ZoneJS tests and logic should be cleaned up to not contain any IE11 specific code. There are some IE11-specific tests for example though (these don't even run at this point anymore -- worth mentioning)

### Please provide a link to a minimal reproduction of the bug

_No response_

### Please provide the exception or error you saw

_No response_

### Please provide the environment you discovered this bug in (run `ng version`)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by JeanMeche**
This commit removes the support for legacy browsers.

BREAKING CHANGE: IE/Non-Chromium Edge are not supported anymore.

Fixes #46773

### CODE REVIEW

**Review by josephperrott — APPROVED** [review]
LGTM

File: packages/zone.js/sauce.conf.js [hunk 1]
```diff
@@ -46,18 +46,6 @@ module.exports = function (config, ignoredLaunchers) {
     // 'SL_IOS9': {base: 'SauceLabs', browserName: 'iphone', platform: 'OS X 10.10', version:
     // '9.3'},
     'SL_IOS10': {base: 'SauceLabs', browserName: 'iphone', platform: 'OS X 10.10', version: '10.3'},
```
Comment (AndrewKushnir) [inline packages/zone.js/sauce.conf.js:48]: nit for some followup PRs: we may need to update the list of supported browsers and corresponding versions, it looks like some of the browsers in the list are pretty old.

**Review by AndrewKushnir — APPROVED** [review]
🎉

File: packages/zone.js/sauce.conf.js (hunk 1 above)
Comment (JeanMeche) [reply packages/zone.js/sauce.conf.js:48]: I'm not even sure this config is used. 
It seems the zone.js saucelabs tests run the same versions as the FW (Android 11/12). See https://github.com/angular/angular/actions/runs/9021600666/job/24789278467?pr=55745

Comment (AndrewKushnir) [reply packages/zone.js/sauce.conf.js:48]: Sounds good, we can revisit testing config and update it in a followup PR.

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:632310648:BASE:632310673:1715299129273:ebb726a5).

**Comment by AndrewKushnir** [comment]
Closing for now, since we won't be able to land this PR with the current scope of changes (since they break code in Google's codebase). We should probably still land parts of this PR (to update browser versions we use for testing, etc) as separate PRs.

**Pull request closed without merging**